backend/src/cache/server/mutation.go

```diff
@@ -218,11 +220,10 @@ func isKFPArgoPod(annotations *map[string]string, podName string) bool {
 		return false
 	}
 	// is KFP pod or not
-	for k := range *annotations {
-		if strings.Contains(k, KFPAnnotation) {
-			return true
-		}
+	if v, exists := (*annotations)[KFPAnnotationKeyCacheEnabled]; exists && v == KFPAnnotationValueCacheEnabled {
```

Comment:
What do you thonk about moving this check [early](https://github.com/kubeflow/pipelines/pull/3368/files#diff-8264bd688f4e96b208c60c26d2a5cd72R77) in the pod object processing so that it's obvious that the non cache-enabled pods are skipped.

P.S. There might be a way to pre-filter pods based on labels in the admission controller configuration.

Reply:
That would be a great idea. Found reference here: https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/#matching-requests-objectselector. Is this what you are proposing?

Reply:
Yes.
P.S. I'm not asking to implement that proposal in this PR.

Reply:
I think that is not much change. How about we hold this PR for now and wait after the changing label pr check in.

Reply:
Sounds good.
BTW having the label in the webhook config also makes it configurable from the outside instead of being hardcoded in the code of the service.